app/models/stat.rb

```diff
@@ -0,0 +1,15 @@
+class Stat < ApplicationRecord
+  belongs_to :dragon
+
+  def add_bonus_from_perception(quantity)
+    Integer(quantity * (1 + (perception / 100.00)))
+  end
+
+  def find_random_resource
+    chance = rand(1..100)
+    return nil if luck < chance
+
+    choose_resource = rand(1..ResourceType.all.size)
+    ResourceType.find_by(id: choose_resource)
```

Comment:
To nie będzie działać.
Załóżmy że mamy 5 typów zasobów więc `choose_resource` będzie z zakresu [1,5]. ID wcale nie muszą być z tego zakresu. Załóżmy, że na początku istnienia gry zostały dodane typy zasobów i one rzeczywiście dostały ID zaczynając od 1, 2 itd., ale potem zostały usunięte. Możemy mieć ResourceTypes o ID np.  11, 12, 15, 16, 17. Będzie Ich 5, ale ID nie mieszczą się w [1,5]. 

`ResourceType.offset(choose_resource).first` - coś takiego powinno działać ale nie testowałem
ew wolniejsza wersja
`ResourceType.all.to_a[choose_resource+1]` - ta pobiera wszystkie zasoby z bazy (co może być obciązające)

Reply:
Można też zrobić `ResourceTyoe.all.sample`